### PR TITLE
[libc] Fix missing default value for errno config

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
The configs all need default values which targets then override. This
one was an empty string which made the logic report an error. The only
reason it wasn't a build failure was because of a stray `:`.

### CODE REVIEW

File: libc/config/config.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "errno": {
     "LIBC_CONF_ERRNO_MODE": {
-      "value": "",
+      "value": "LIBC_ERRNO_MODE_DEFAULT",
```
Comment (lntue) [inline libc/config/config.json:4]: Can you add this option to the docstring below?

File: libc/src/errno/libc_errno.cpp [hunk 2]
```diff
@@ -23,7 +25,8 @@
 // fullbuild mode, effectively the same as `LIBC_ERRNO_MODE_EXTERNAL`.
 #define LIBC_ERRNO_MODE_SYSTEM 5
 
-#ifndef LIBC_ERRNO_MODE
+#if !defined(LIBC_ERRNO_MODE) || LIBC_ERRNO_MODE == LIBC_ERRNO_MODE_DEFAULT
```
Comment (lntue) [inline libc/src/errno/libc_errno.cpp:28]: Can you add `LIBC_ERRNO_MODE_DEFAULT` to the `#error` message below?

File: libc/config/config.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
       "doc": "The implementation used for errno, acceptable values are LIBC_ERRNO_MODE_UNDEFINED, LIBC_ERRNO_MODE_THREAD_LOCAL, LIBC_ERRNO_MODE_SHARED, LIBC_ERRNO_MODE_EXTERNAL, and LIBC_ERRNO_MODE_SYSTEM."
```
Comment (jhuber6) [inline libc/config/config.json:5]:
```suggestion
      "doc": "The implementation used for errno, acceptable values are LIBC_ERRNO_DEFAULT, LIBC_ERRNO_MODE_UNDEFINED, LIBC_ERRNO_MODE_THREAD_LOCAL, LIBC_ERRNO_MODE_SHARED, LIBC_ERRNO_MODE_EXTERNAL, and LIBC_ERRNO_MODE_SYSTEM."
```

File: libc/config/config.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
   "errno": {
     "LIBC_CONF_ERRNO_MODE": {
-      "value": "",
-      "doc": "The implementation used for errno, acceptable values are LIBC_ERRNO_MODE_UNDEFINED, LIBC_ERRNO_MODE_THREAD_LOCAL, LIBC_ERRNO_MODE_SHARED, LIBC_ERRNO_MODE_EXTERNAL, and LIBC_ERRNO_MODE_SYSTEM."
+      "value": "LIBC_ERRNO_MODE_DEFAULT",
+      "doc": "The implementation used for errno, acceptable values are LIBC_ERRNO_DEFAULT, LIBC_ERRNO_MODE_UNDEFINED, LIBC_ERRNO_MODE_THREAD_LOCAL, LIBC_ERRNO_MODE_SHARED, LIBC_ERRNO_MODE_EXTERNAL, and LIBC_ERRNO_MODE_SYSTEM."
```
Comment (jhuber6) [inline libc/config/config.json:5]:
```suggestion
      "doc": "The implementation used for errno, acceptable values are LIBC_ERRNO_MODE_DEFAULT, LIBC_ERRNO_MODE_UNDEFINED, LIBC_ERRNO_MODE_THREAD_LOCAL, LIBC_ERRNO_MODE_SHARED, LIBC_ERRNO_MODE_EXTERNAL, and LIBC_ERRNO_MODE_SYSTEM."
```

File: libc/src/errno/libc_errno.cpp [hunk 5]
```diff
@@ -23,15 +25,17 @@
...
+#undef LIBC_ERRNO_MODE
 #if defined(LIBC_FULL_BUILD) || !defined(LIBC_COPT_PUBLIC_PACKAGING)
 #define LIBC_ERRNO_MODE LIBC_ERRNO_MODE_THREAD_LOCAL
 #else
 #define LIBC_ERRNO_MODE LIBC_ERRNO_MODE_SYSTEM
 #endif
 #endif // LIBC_ERRNO_MODE
 
-#if LIBC_ERRNO_MODE != LIBC_ERRNO_MODE_UNDEFINED &&                            \
+#if LIBC_ERRNO_MODE != LIBC_ERRNO_MODE_DEFAULT &&                              \
```
Comment (lntue) [inline libc/src/errno/libc_errno.cpp:37]: add `LIBC_ERRNO_MODE_DEFAULT` to L44 below.

**Review by lntue — APPROVED** [review]

**Pull request merged**